parsec/core/fs/exceptions.py

```diff
@@ -42,134 +42,132 @@ def __str__(self):
 
 
-class FSCrossDeviceError(FSError):
+class FSCrossDeviceError(FSOperationLocalError):
     ERRNO = errno.EXDEV
     NTSTATUS = ntstatus.STATUS_NOT_SAME_DEVICE
 
 
-class FSFileExistsError(FSError, FileExistsError):
+class FSFileExistsError(FSOperationLocalError, FileExistsError):
     ERRNO = errno.EEXIST
     NTSTATUS = ntstatus.STATUS_OBJECT_NAME_COLLISION
 
 
-class FSIsADirectoryError(FSError, IsADirectoryError):
+class FSIsADirectoryError(FSOperationLocalError, IsADirectoryError):
     ERRNO = errno.EISDIR
     NTSTATUS = ntstatus.STATUS_FILE_IS_A_DIRECTORY
 
 
-class FSDirectoryNotEmptyError(FSError):
+class FSDirectoryNotEmptyError(FSOperationLocalError):
     ERRNO = errno.ENOTEMPTY
     NTSTATUS = ntstatus.STATUS_DIRECTORY_NOT_EMPTY
 
 
-class FSInvalidFileDescriptor(FSError):
+class FSInvalidFileDescriptor(FSOperationLocalError):
     ERRNO = errno.EBADF
     NTSTATUS = ntstatus.STATUS_INVALID_HANDLE
 
 
-class FSNetworkError(FSError):
-    ERRNO = errno.EHOSTUNREACH
-    NTSTATUS = ntstatus.STATUS_HOST_UNREACHABLE
-
-
-class FSEntryNotFound(FSError):
-    ERRNO = errno.ENOENT
-    NTSTATUS = ntstatus.STATUS_OBJECT_NAME_NOT_FOUND
-
-
-class FSInvalidArgumentError(FSError):
+class FSInvalidArgumentError(FSOperationLocalError):
     ERRNO = errno.EINVAL
     NTSTATUS = ntstatus.STATUS_INVALID_PARAMETER
 
 
-class FSInternalError(FSError):
-    def __init__(self, *args):
-        super(Exception, self).__init__(*args)
-
-    def __str__(self):
-        return super(Exception, self).__str__()
-
-
-# Protocol errors
-
-
-class FSValidationError(FSInternalError):
-    pass
+# Operation remote errors
 
 
-class FSPackingError(FSInternalError):
-    pass
-
-
-# Remote errors
+class FSBackendOfflineError(FSOperationRemoteError):
+    ERRNO = errno.EHOSTUNREACH
+    NTSTATUS = ntstatus.STATUS_HOST_UNREACHABLE
 
 
-class FSRemoteManifestNotFound(FSInternalError):
+class FSRemoteManifestNotFound(FSOperationRemoteError):
     pass
 
 
 class FSRemoteManifestNotFoundBadVersion(FSRemoteManifestNotFound):
     pass
 
 
-class FSRemoteBlockNotFound(FSInternalError):
+class FSRemoteBlockNotFound(FSOperationRemoteError):
     pass
 
 
-class FSRemoteSyncError(FSInternalError):
+class FSRemoteSyncError(FSOperationRemoteError):
     pass
 
 
-class FSBadEncryptionRevision(FSInternalError):
+class FSBadEncryptionRevision(FSOperationRemoteError):
     pass
 
 
-# Local miss errors
-
-
-class FSLocalMissError(FSInternalError):
-    def __init__(self, id: EntryID):
-        super().__init__(id)
-        self.id = id
-
-
-class FSWorkspaceNotFoundError(FSLocalMissError):
+class FSSharingNotAllowedError(FSOperationRemoteError):
     pass
 
 
-# Connection errors
-
-
-class FSBackendOfflineError(FSNetworkError):
-    pass
-
-
-# Rights errors
-
-
-class FSSharingNotAllowedError(FSPermissionError):
-    pass
-
-
-class FSWorkspaceNoAccess(FSPermissionError):
+class FSWorkspaceNoAccess(FSOperationRemoteError, PermissionError):
```

Comment:
Those exceptions may be used locally:
https://github.com/Scille/parsec-cloud/blob/033324ca8c2e64584906e790a33ec4a85f6b2b96/parsec/core/fs/workspacefs/entry_transactions.py#L40-L42

Reply:
582a7721e50243d05fc2082258ad979863844dcb